src/com/google/javascript/jscomp/CommandLineRunner.java

```diff
@@ -1931,6 +1939,11 @@ protected CompilerOptions createOptions() {
       options.setPropertyRenaming(PropertyRenamingPolicy.OFF);
     }
 
+    if (flags.instrumentCodeParsed == InstrumentOption.PRODUCTION &&
```

Comment:
Also check the other way around? If report is paramMappingOutputFile is provided while `PRODUCTION` is not enabled?